arcgis-ios-sdk-samples/Layers/ArcGIS tiled layer (tile cache)/TilePackagesListViewController.swift

```diff
@@ -71,40 +71,40 @@ class TilePackagesListViewController: UIViewController, UITableViewDataSource, U
         }
     }
     
-    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCellWithIdentifier("TilePackageCell")!
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "TilePackageCell")!
         
-        if indexPath.section == 0 {
-            cell.textLabel?.text = self.extractName(self.bundleTPKPaths[indexPath.row])
+        if (indexPath as NSIndexPath).section == 0 {
+            cell.textLabel?.text = self.extractName(self.bundleTPKPaths[(indexPath as NSIndexPath).row])
         }
         else {
-            cell.textLabel?.text = self.extractName(self.documentTPKPaths[indexPath.row])
+            cell.textLabel?.text = self.extractName(self.documentTPKPaths[(indexPath as NSIndexPath).row])
         }
         
         return cell
     }
     
-    func tableView(tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
         return section == 0 ? "From the bundle" : "From the documents directory"
     }
     
     //MARK: - UITableViewDelegate
     
-    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         var path = ""
-        if indexPath.section == 0 {
-            path = self.bundleTPKPaths[indexPath.row]
+        if (indexPath as NSIndexPath).section == 0 {
+            path = self.bundleTPKPaths[(indexPath as NSIndexPath).row]
         }
         else {
-            path = self.documentTPKPaths[indexPath.row]
+            path = self.documentTPKPaths[(indexPath as NSIndexPath).row]
         }
         self.delegate?.tilePackagesListViewController(self, didSelectTPKWithPath: path)
     }
     
-    func extractName(path:String) -> String {
-        var index = path.rangeOfString("/", options: .BackwardsSearch, range: nil, locale: nil)?.startIndex
-        index = index?.advancedBy(1)
-        let name = path.substringFromIndex(index!)
+    func extractName(_ path:String) -> String {
```

Comment:
similar to other comments above about having argument name to clarify call site usage